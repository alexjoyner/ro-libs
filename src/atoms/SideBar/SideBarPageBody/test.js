import React from 'react';
import { shallow } from 'enzyme';
import BasicSideBarPageBody, { HeaderSideBarPageBody } from './';

describe('SideBarPageBody component', () => {
  let component;
  describe('Basic Version', () => {
    beforeEach(() => {
      component = shallow(<BasicSideBarPageBody />);
    });
    it('Should render without exploding', () => {
      expect(component).toBeDefined();
    });
    it('Should match snapshot', () => {
      expect(component).toMatchSnapshot();
    });
    it('Should have default styles', () => {
      expect(component).toHaveStyleRule('margin-left', '300px');
      expect(component).toHaveStyleRule('min-height', '10px');
    });
  });
  describe('with Header Version', () => {
    beforeEach(() => {
      component = shallow(<HeaderSideBarPageBody />);
    });
    it('Should render without exploding', () => {
      expect(component).toBeDefined();
    });
    it('Should match snapshot', () => {
      expect(component).toMatchSnapshot();
    });
    it('Should have default styles', () => {
      expect(component).toHaveStyleRule('margin-top', '56px');
    });
  });
});
