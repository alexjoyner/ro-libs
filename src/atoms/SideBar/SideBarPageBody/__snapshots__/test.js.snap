// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBarPageBody component Basic Version Should match snapshot 1`] = `
.c0 {
  -webkit-transition: 100ms linear;
  transition: 100ms linear;
  margin-left: 300px;
  min-height: 10px;
}

<SideBarPageBody__BasicSideBarPageBody
  sideBarShown={true}
>
  <StyledComponent
    forwardedClass={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "SideBarPageBody__BasicSideBarPageBody-sc-1qqh35b-0",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "transition:100ms linear;margin-left:",
            [Function],
            ";min-height:10px;",
          ],
        },
        "displayName": "SideBarPageBody__BasicSideBarPageBody",
        "render": [Function],
        "styledComponentId": "SideBarPageBody__BasicSideBarPageBody-sc-1qqh35b-0",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    sideBarShown={true}
  >
    <div
      className="c0"
    />
  </StyledComponent>
</SideBarPageBody__BasicSideBarPageBody>
`;

exports[`SideBarPageBody component with Header Version Should match snapshot 1`] = `
.c0 {
  -webkit-transition: 100ms linear;
  transition: 100ms linear;
  margin-left: 0px;
  min-height: 10px;
  margin-top: 56px;
}

<SideBarPageBody__HeaderSideBarPageBody>
  <StyledComponent
    forwardedClass={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "SideBarPageBody__HeaderSideBarPageBody-sc-1qqh35b-1",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "transition:100ms linear;margin-left:",
            [Function],
            ";min-height:10px;",
            "margin-top:56px;",
          ],
        },
        "displayName": "SideBarPageBody__HeaderSideBarPageBody",
        "render": [Function],
        "styledComponentId": "SideBarPageBody__HeaderSideBarPageBody-sc-1qqh35b-1",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="c0"
    />
  </StyledComponent>
</SideBarPageBody__HeaderSideBarPageBody>
`;
